Add optional code for well testing

joakim-hove opened the pull request:
A refactoring of the wells in the opm-common repository has started:  https://github.com/OPM/opm-common/issues/624

The intention is to create a new well implementation side by side with the existing old implementation, and then switch over to use the new implementation when everything is ready. The new well code will be tested the normal way with unit tests, but the well code is complex - and it would be valuable to have some integration testing as early as possible - this PR is an attempt to address that.

The PR is based on a cmake option `ENABLE_WELL_TESTING` which will define the preprocessor symbol `WELL_TESTING` - and then if the `WELL_TESTING` symbol is defined a method `checkWells()` is added to the `Schedule` object - the `checkWells()` method will be called at the end of the `Schedule` constructor.

The cmake switch `ENABLE_WELL_TESTING` defaults to OFF, I will of course compile with the switch set to ON - ~~I also intend to submit a PR which will pass `-DENABLE_WELL_TESTING=ON` when configuring opm-common in the integration testing~~, and also hope other developers will build with this setting. The downside of beeing a volunteer integration tester is that some extra code will run as part of the `Schedule` construction, and *iff* there is a discrepancy in the results between the new well code and the old well code the program execution will stop. **Update**: iff there is an error - there will be a ParseContext switch to determine exactly how the error should be handeled.

The switch will of course be removed again when the work is complete - I guesstimate to months of work. If there are other suggestions to how this should be achieved I am all ears!

joakim-hove commented:
jenkins build this please

joakim-hove commented:
jenkins build this please

bska commented on src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp:
```diff
@@ -104,6 +104,11 @@ namespace Opm {
             iterateScheduleSection( parseContext, errors, SCHEDULESection( deck ), grid, eclipseProperties );
+
+#ifdef CHECK_WELLS
+        check_wells();
```
Is this supposed to say `checkWells()`?

joakim-hove replied:
Yes - thank you.

joakim-hove commented:
jenkins build this please

joakim-hove commented:
I have addressed the comments from review, and also added a `(ParseContext, ErrorGuard)`pair to the `Schedule::checkWells()` method - to ensure that it is possible to opt out of application stop when/if an inconsistency is encountered.

Will merge when Jenkins is green.
